source/blog/2017-10-10-glimmer-progress-report.md

```diff
@@ -0,0 +1,645 @@
+
+If you notice that a lot of the items I describe below are performance-related,
+that is at least partly due to our product's near-maniacal focus on mobile
+load times. We are extremely excited about some of the recent breakthroughs
+we've made and have enjoyed proving out some of the more esoteric ideas in a
+real app.
+
+## What's New in Glimmer
+
+### `<my-component />` → `<Component />`
+
+One of the most eagerly-awaited features of Glimmer.js is "angle bracket
+components," or components that you invoke `<like-this />` instead of
+`{{like-this}}`. I personally really like this syntax because it visually
+disambiguates components from the dynamic data that flows through them.
+It also unifies the attribute syntax between HTML and components:
+
+```hbs
+{{! Ember }}
+{{my-button title=title label=(t "Do Something")}}
+
+{{! HTML }}
+<button title={{title}} label={{t "Do Something"}}></button>
+
+{{! Glimmer.js }}
+<my-button @title={{title}} @label={{t "Do Something"}} />
+```
+
+This syntax is how Glimmer.js works today. However, the more we discussed the
+design and started using it in real projects, the more we believed that this
+exact API was flawed and needed to be rethought.
+
+First, a short history lesson. When we introduced components in Ember (back
+before Ember 1.0!), we required them to include a dash (`-`) in their name. This
+rule came from the then-new [Custom Elements spec][custom-elements], a key part
+of the Web Components API.
+
+[custom-elements]: https://www.w3.org/TR/custom-elements/#custom-element-conformance
+
+Web Components have the problem of needing to disambiguate between built-in elements
+and custom elements. What happens if I make a component called `<vr></vr>` and later
+the browser adds a built-in Virtual Reality element with the same name?
+
+The compromise was that custom elements must have a dash, keeping single-word
+elements reserved for future versions of the HTML standard.
+
+Of course, Ember components don't have the same problem, because they use
+`{{`/`}}` as delimeters instead of `<`/`>`. Nonetheless, we preemptively adopted
+this constraint because we assumed Web Components were going to take the world
+by storm and at some point we would need to migrate Ember components to Web
+Components.
+
+As time has passed, though, it's become increasingly clear that the use cases
+served by Web Components, wonderful as they are, do not have the full set of
+functionality to replace everything that an Ember component (or React component,
+etc.) needs to do.
+
+Meanwhile, the more components I write, the more grating the naming restriction
+feels. I hate the cognitive overhead of having to invent silly names like
+`{{x-button}}` when a single word would be much more descriptive.
+
+Unfortunately, this puts us in a bit of a pickle:
+
+1. We want to drop the annoying `dasherized-component` requirement.
+2. We want to adopt `<angle-brackets>` syntax for components, but now we have
```

Comment:
Maybe "We want to adopt angle syntax for componments". Item 1 in this list says "we want to drop dash-erized" and the second says "we want to adopt <dash-erized" which is confusing. I think you should focus the second point on the `<` character and drop the string `angle-brackets`.

Reply:
Great catch, thank you!